inc/composer/class-command.php

```diff
@@ -654,14 +705,175 @@ protected function db( InputInterface $input, OutputInterface $output ) {
+					$output->writeln( '<error>This command is only supported on macOS, Linux, and Windows x64, install `mkcert` manually for other systems.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+
+				$binary = "mkcert-$mkcert_version-$binary_arch";
+				$mkcert = 'vendor/mkcert';
+
+				// Check if mkcert is installed globally already, bail if so.
+				$version = trim( shell_exec( 'mkcert -version' ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed globally already</error>" );
+					return 1;
+				}
+
+				// Check if mkcert is installed locally already, bail if so.
+				$version = trim( shell_exec( "$mkcert -version" ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed locally already</error>" );
+					return 1;
+				}
+
+				exec( "curl -o $mkcert -L https://github.com/FiloSottile/mkcert/releases/download/$mkcert_version/$binary", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( '<error>Could not download mkcert binary, try using sudo or manually installing mkcert.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert $mkcert_version was downloaded.</info>" );
+
+				chmod( $mkcert, 0755 );
+
+				exec( "$mkcert -version", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( '<error>Could not launch mkcert binary, try manually installing mkcert.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+				$output->writeln( "<info>mkcert $mkcert_version was installed.</info>" );
+
+				// Setup and accept the root certificate.
+				exec( "$mkcert -install", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( '<error>Could not setup `mkcert` properly, try manually installing it.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+
+				$output->writeln( '<info>mkcert root CA was installed and accepted successfully.</info>' );
+				break;
+			case 'generate':
+				// TODO figure out how to programmatically detect the domains to use.
+				$domains = $input->getArgument( 'options' )[1] ?? '*.altis.dev';
+
+				exec( "$mkcert -cert-file vendor/ssl-cert.pem -key-file vendor/ssl-key.pem $domains", $dummy, $result );
+
+				if ( $result ) {
+					$output->writeln( '<error>Could not generate certificates! Try generating them manually using `mkcert`.</error>' );
+					$output->writeln( "<error>Command should be: 'mkcert -cert-file vendor/ssl-cert.pem -key-file vendor/ssl-key.pem $domains'</error>" );
+					return 1;
+				}
+
+				$output->writeln( '<info>Generated SSL certificate successfully.</info>' );
```

Comment:
Will the proxy need to be restarted here? Thinking it might as it's traefik 1.7.

Reply:
Not 100% sure I tested that scenario, needs to be tested.

Reply:
It needed to be restarted indeed, added logic to restart the proxy service if running, after the generation process.